pkg/types/baremetal/platform.go

```diff
@@ -61,4 +61,14 @@ type Platform struct {
 
 	// DNSVIP is the VIP to use for internal DNS communication
 	DNSVIP string `json:"dnsVIP"`
+
+	// BootstrapMachineOSImage is a URL to override the default machine OS image
+	// for the bootstrap node.
+	// +optional
```

Comment:
1) there should be a validation that checks that this value is a valid URL (hopefully http/https),
2) I think by default when we used rhcos.json we ensure that the checksum is valid. So we should enforce that the url have similar query parameter such that the caching code checks the sha. clarity in comment regarding this and also a validation for the same should be added.

also please add unit tests for these validations.

Reply:
agreed - @kirankt you may find the following links helpful as we had to add similar validation to pr #2449

https://github.com/openshift/installer/blob/master/pkg/validate/validate.go#L141

https://github.com/openshift/installer/pull/2449

Reply:
+1. Working on adding this.

Reply:
>     2. I think by default when we used rhcos.json we ensure that the checksum is valid. So we should enforce that the url have similar query parameter such that the caching code checks the sha. clarity in comment regarding this and also a validation for the same should be added.

I think this will become burdensome for a user to specify the checksum with the URL. Since the user is manually specifying the URL, I think it's up to them to determine if it's valid. Besides the cache code doesn't require the checksum. Its skipping validation if the hash is not present.

Reply:
> > ```
> > 2. I think by default when we used rhcos.json we ensure that the checksum is valid. So we should enforce that the url have similar query parameter such that the caching code checks the sha. clarity in comment regarding this and also a validation for the same should be added.
> > ```
> 
> I think this will become burdensome for a user to specify the checksum with the URL. Since the user is manually specifying the URL, I think it's up to them to determine if it's valid. Besides the cache code doesn't require the checksum. Its skipping validation if the hash is not present.

After discussing this internally, we will ensure that the URI contains the sha256 parameter. Update coming.

Reply:
Done. Added validation and tests to chech for valid URL and sha256 checksum.